charts/gateway-helm/templates/envoy-gateway-hpa.yaml

```diff
@@ -0,0 +1,44 @@
+{{- if .Values.autoscaling.enabled -}}
+apiVersion: {{ ternary "autoscaling/v2" "autoscaling/v2beta2" (.Capabilities.APIVersions.Has "autoscaling/v2") }}
+kind: HorizontalPodAutoscaler
+metadata:
+  {{- with .Values.autoscaling.annotations }}
+  annotations: {{ toYaml . | nindent 4 }}
+  {{- end }}
+  labels:
+    control-plane: envoy-gateway
+  {{- include "eg.labels" . | nindent 4 }}
+  name: envoy-gateway
+  namespace: '{{ .Release.Namespace }}'
+spec:
+  scaleTargetRef:
```

Comment:
i don't think this is the target for HPA.

ref: https://github.com/envoyproxy/gateway/issues/703#issuecomment-1827083833